src/Host/Client/Impl/Host/RHost.cs

```diff
@@ -37,7 +41,12 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator, IRBlobSe
         private volatile Task _runTask;
         private volatile Task<REvaluationResult> _cancelEvaluationAfterRunTask;
         private volatile Task<ulong> _cancelCreateBlobAfterRunTask;
-        private volatile Task<byte[]> _cancelGetBlobAfterRunTask;
+        private volatile Task<byte[]> _cancelBlobReadAllAfterRunTask;
+        private volatile Task<byte[]> _cancelBlobReadAfterRunTask;
```

Comment:
You don't need to have so many of them - they all are identical except for the type argument of the task. So you need to add only one more.
Btw, why `CreateBlobAsync` returns `ulong`, while other methods return `long`?